java/code/src/com/suse/manager/reactor/messaging/ImageDeployedEventMessageAction.java

```diff
@@ -75,6 +76,11 @@ public void execute(EventMessage msg) {
             LOG.info("System image of minion id '" + m.getId() + "' has changed. Re-applying activation key," +
                     " subscribing to channels and executing post-registration tasks.");
             ValueMap grains = imageDeployedEvent.getGrains();
+
+            grains.getOptionalAsString("osarch").ifPresent(osarch -> {
+                m.setServerArch(ServerFactory.lookupServerArchByLabel(osarch + "-redhat-linux"));
```

Comment:
Will this never happen for Ubuntu/Debian?

Reply:
We support Kiwi images only on SUSE.
Is there any possibility that somebody needs Ubuntu/Debian images?

Reply:
ok, just wanted to be sure